### ISSUE TITLE
--balance_classes computes class weights but never applies them to the loss function

### ISSUE THREAD

**Issue by lisasterionella** [kaechb/lit_ecology_classifier#2]
Hi, thank you for this package, it's been very useful for our plankton classification project.

While reading through the training code to understand how `--balance_classes` works, I noticed that the computed class weights don't appear to be passed to the loss function.

In `main.py`, when `--balance_classes` is set, the weights are computed like this:

    if args.balance_classes:
        args.class_weights = calculate_class_weights(datamodule)
    else:
        args.class_weights = None

`calculate_class_weights` (in `helpers/calc_class_weights.py`) correctly computes per-class weights:

    weights = 1 / torch.bincount(torch.cat(labels)).float().sqrt()

However, in `models/model.py`, the loss function is initialized without ever referencing `class_weights`:

    self.loss = torch.nn.CrossEntropyLoss() if not "loss" in list(self.hparams) or not self.hparams.loss=="focal" else FocalLoss(alpha=None, gamma=1.75)

I searched the whole package for `class_weights` and `CrossEntropyLoss`/`weight=` and couldn't find anywhere else where `args.class_weights` is actually used. So as far as I can tell, `--balance_classes` currently has no effect on training, despite computing the weights correctly.

Is this expected, or is this a bug? If it's the latter, would something like this in `model.py` be the intended fix?

    self.loss = torch.nn.CrossEntropyLoss(weight=self.hparams.get("class_weights")) if ...

I am a beginner with code and just wanted to check with you first in case I'm missing something.

Thanks!
